### DISCUSSION TITLE
Service Cloud Connector data sync not fully working?

### DISCUSSION THREAD

**Discussion by RostyslavKostyk** [SalesforceCommerceCloud/sfcc-ci#416] [Team Posts]
Hello,
I've installed newest version of Service Cloud Connector (2.0) and in release notes it is said "Out of the box Order Line Items, Shipments and Payment Information related objects", does this mean they should be exported along with orders out of the box? Because right now what we have is orders/customers are getting exported just fine, but related Order Line Items, Payment information or Shipments are not populated, as well as Addresses for customers. Is it something we've missed during installation process or there's some additional things required?
Thanks alot.

**Comment by qzhao-sfcc** [comment]
Hi Rostyslav,

So you are right, only order header information are initially exported to
Service Cloud to create order. However, there is a callback from service
cloud afterwards to retrieve order details via OCAPI API. This process is
to improve replication efficiency in case of big volume or data and avoid
reaching API limits from service cloud side.

To get order details, you can check more details in following class:
⁨service-cloud-connector-release/⁨b2c⁩/⁨sfsc⁩/⁨src⁩/⁨classes⁩/SCCOrderImpl.cls
 -> continueWithObject()
*(remember to uncomment line upsertProductShipping(record, jsonMap, logger)
to update order details)*
⁨service-cloud-connector-release⁩/b2c⁩/⁨sfsc⁩/src⁩/classes⁩/SCCCommerceCloudConnect.cls
-> getOrderDetails()

Same logic applies to customer replication and hope this answers your
question.

Best Regards,
Qingyang

On Thu, 27 Jun 2019 at 16:22, Rostyslav Kostyk <notifications@github.com>
wrote:

> Hello,
> I've installed newest version of Service Cloud Connector (2.0) and in
> release notes it is said "Out of the box Order Line Items, Shipments and
> Payment Information related objects", does this mean they should be
> exported along with orders out of the box? Because right now what we have
> is orders/customers are getting exported just fine, but related Order Line
> Items, Payment information or Shipments are not populated, as well as
> Addresses for customers. Is it something we've missed during installation
> process or there's some additional things required?
> Thanks alot.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/SalesforceCommerceCloud/teams/community/discussions/64?email_source=notifications&email_token=ALXI2IAK5SGHSXO74ZELCHLP4TLKTA5CNFSM4H35DZAKYY3PNVWWK3TUL52HS4DFVZCGS43DOVZXG2LPNZIG643UVJRW63LNMVXHIX3JMTHAAAIXDI>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALXI2IDJQWWD7DYJTD6CUJLP4TLKTANCNFSM4H35DZAA>
> .
>


-- 
*Qingyang Zhao*
Senior Technical Consultant
Commerce Cloud B2C Services | CSG EMEA Services

Salesforce Tower London

110 Bishopsgate

London, EC2N 4AY, UK
Mobile: +44 7584 545 215

<http://www.salesforce.com/signature>

**Comment by pravngaur** [comment]
As Quin mentioned, SC connector posts very tiny information in SC, which then gets queued for processing. Upon execution SC calls back the SFCC OCAPI to get the full order details - so if the OCAPI(s) is not configured properly in SFCC, the 2nd call will result in error & then you would see the partial order details in SC.

**Comment by RostyslavKostyk** [comment]
thanks for clarification guys!

**Comment by shayunBBD** [comment]
Hello @RostyslavKostyk , Can you share what did you do?
